integreat_cms/api/v3/socialmedia_headers.py

```diff
@@ -0,0 +1,185 @@
+
+    from django.http import HttpRequest, HttpResponse
+
+logger = logging.getLogger(__name__)
+
+
+def socialmedia_headers(
+    request: HttpRequest, path: Optional[str] = None
+) -> HttpResponse:
+    """
+    Detects the path context and renders the socialmedia headers.
+    Available contexts are:
+    - root e.g. /, /landing, /disclaimer, /landing/de
+    - region root e.g. /testumgebung, /testumgebung/de, /testumgebung/de/events
+    - page e.g. /testumgebung/de/willkommen, /testumgebung/de/events/event
+
+    :param request: The request that has been sent to the Django server
+    :param path: The path of the page the social meddia headers should be created for
+
+    :return: HTML meta headers required by social media platforms
+    """
+    slugs = list(filter(bool, path.split("/"))) if path else []
+
+    potential_region_slug = slugs[0] if len(slugs) > 0 else None
+    region = (
+        Region.objects.filter(slug=potential_region_slug).first()
+        if potential_region_slug
+        else None
+    )
+
+    potential_language_slug = slugs[1] if len(slugs) > 1 else None
+    language = Language.objects.filter(
+        slug=potential_language_slug
+    ).first() or Language.objects.get(slug="de")
+    if not region:
+        return root_socialmedia_headers(request, language, path or "")
+
+    if len(slugs) < 3:
+        return region_socialmedia_headers(request, region, path or "")
+
+    return page_socialmedia_headers(request, region, language, "/".join(slugs[2:]))
+
+
+def root_socialmedia_headers(
+    request: HttpRequest, language: Language, path: str
+) -> HttpResponse:
+    """
+    Renders the socialmedia headers for a root page
+
+    :param request: The request that has been sent to the django server
+    :param language: The language of the page
+    :param path: The path to the page
+
+    """
+    title = language.socialmedia_webapp_title
+    url = f"{settings.WEBAPP_URL}/{path}"
+
+    excerpt = (
+        (
+            "hallo aschaffenburg is your digital companion for the town of Aschaffenburg. Here"
+            " you can find local information, advice centres and services."
+        )
+        if settings.BRANDING == "aschaffenburg"
+        else (
+            settings.BRANDING_TITLE + " is your digital companion for Germany. Here"
+            " you can find local information, advice centres and services."
+        )
+    )
+    return render_socialmedia_headers(request, title, excerpt, url)
+
+
+def region_socialmedia_headers(
+    request: HttpRequest, region: Region, path: str
+) -> HttpResponse:
+    """
+    Renders the social media headers for a root region page
+
+    :param request: The request that has been sent to the Django Server
+    :param region: The region the request refers to
+    :param path: The path to the page
+    """
+    title = f"{region.name} | {settings.BRANDING_TITLE}"
+    url = f"{settings.WEBAPP_URL}/{path}"
+    return render_socialmedia_headers(request, title, None, url)
```

Comment:
Omit the excerpt for these pages. Even though, I'm sure google will handle this correctly and crawl the content, I'm actually not sure about `TwitterBot`, `LinkedInBot`, etc. ...